tests/sklearn/test_sklearn_autolog.py

```diff
@@ -1060,6 +1060,25 @@ def test_autolog_does_not_throw_when_infer_signature_fails():
     assert "signature" not in model_conf.to_dict()
 
 
+def test_autolog_does_not_warn_when_model_has_transform_function():
+    X, y = get_iris()
+
+    with mlflow.start_run() as run, mock.patch("mlflow.sklearn._logger.warning") as mock_warning:
+        mlflow.sklearn.autolog(log_input_examples=True, log_model_signatures=True)
+        estimators = [
+            ("std_scaler", sklearn.preprocessing.StandardScaler()),
+        ]
+        model = sklearn.pipeline.Pipeline(estimators)
+        model.fit(X, y)
+
+    # Warning not called
+    with pytest.raises(AssertionError, match=r"expected call not found"):
+        mock_warning.assert_called_with("Failed to infer model signature: Failed")
```

Comment:
This is to make sure the warning is not called, so when we call 'assert_called_with' it throws AssertionError. Because mock doesn't have assert_not_called_with method.

Reply:
Got it, I think you can use 

```
assert all("Failed to infer model signature: Failed" in c[0] for c in mock_warning.call_args_list)
```